hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindTargetGreedy.java

```diff
@@ -52,30 +57,19 @@
   private NodeManager nodeManager;
   private ContainerBalancerConfiguration config;
   private Double upperLimit;
-  private TreeSet<DatanodeUsageInfo> potentialTargets;
+  private Collection<DatanodeUsageInfo> potentialTargets;
+  private NetworkTopology networkTopology;
 
   public FindTargetGreedy(
```

Comment:
How about we make `FindTargetGreedy` an abstract class and have two inherited classes FindTargetByUsage and FindTargetByNetworkTopology?

Reply:
sure , will do this